worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,454 @@
+                self.options.final_campaign_boss_campaign_opponents.value = \
+                    slot_data["final_campaign_boss_campaign_opponents"]
+                self.options.fourth_tier_5_campaign_boss_campaign_opponents.value = \
+                    slot_data["fourth_tier_5_campaign_boss_campaign_opponents"]
+                self.options.third_tier_5_campaign_boss_campaign_opponents.value = \
+                    slot_data["third_tier_5_campaign_boss_campaign_opponents"]
+                self.options.number_of_challenges.value = \
+                    slot_data["number_of_challenges"]
+                self.removed_challenges = slot_data["removed challenges"]
+                self.starting_booster = slot_data["starting_booster"]
+                self.starting_opponent = slot_data["starting_opponent"]
+
+        if self.options.structure_deck.current_key == "none":
+            self.is_draft_mode = True
+            boosters = draft_boosters
+            if self.options.campaign_opponents_shuffle.value:
+                opponents = tier_1_opponents
+            else:
+                opponents = draft_opponents
+        else:
+            boosters = booster_packs
+            opponents = tier_1_opponents
+
+        if self.options.structure_deck.current_key == "random_deck":
+            self.options.structure_deck.value = self.random.choice([0, 1, 2, 3, 4, 5])
+        for item in self.options.start_inventory:
+            if item in opponents:
+                self.starting_opponent = item
+            if item in boosters:
+                self.starting_booster = item
+        if not self.starting_opponent:
+            self.starting_opponent = self.random.choice(opponents)
+        self.multiworld.push_precollected(self.create_item(self.starting_opponent))
+        if not self.starting_booster:
+            self.starting_booster = self.random.choice(boosters)
+        self.multiworld.push_precollected(self.create_item(self.starting_booster))
+        banlist = self.options.banlist.value
+        self.multiworld.push_precollected(self.create_item(Banlist_Items[banlist]))
+
+        if not self.removed_challenges:
+            challenge = list((Limited_Duels | Theme_Duels).keys())
+            noc = len(challenge) - max(self.options.third_tier_5_campaign_boss_challenges.value
+                                   if self.options.third_tier_5_campaign_boss_unlock_condition == "challenges" else 0,
+                                   self.options.fourth_tier_5_campaign_boss_challenges.value
+                                   if self.options.fourth_tier_5_campaign_boss_unlock_condition == "challenges" else 0,
+                                   self.options.final_campaign_boss_challenges.value
+                                   if self.options.final_campaign_boss_unlock_condition == "challenges" else 0,
+                                   self.options.number_of_challenges.value)
```

Comment:
Really would like to see this reformatted as well:
```
noc = len(challenge) - max(
    self.options...
    ...
)
```

Then you can give `noc` a more descriptive variable name as well